src/MainWindow.vala

```diff
@@ -94,18 +94,61 @@ public class Maps.MainWindow : Adw.ApplicationWindow {
 
         search_entry.set_key_capture_widget (search_res_popover);
 
+        var explore_source_button = new Gtk.ToggleButton () {
+            action_name = "win.map-source",
+            action_target = Define.MapSource.MAPNIK,
+            child = new Gtk.Image.from_icon_name ("map-tile-explore") {
+                pixel_size = 48
+            }
+        };
+        explore_source_button.add_css_class ("image-button");
+
+        var explore_source_label = new Gtk.Label (_("Explore")) {
+            mnemonic_widget = explore_source_button
+        };
+        explore_source_label.add_css_class (Granite.STYLE_CLASS_SMALL_LABEL);
+
+        var explore_source_box = new Gtk.Box (VERTICAL, 0);
+        explore_source_box.append (explore_source_button);
+        explore_source_box.append (explore_source_label);
+
+        var transit_source_button = new Gtk.ToggleButton () {
+            action_name = "win.map-source",
+            action_target = Define.MapSource.TRANSPORT,
+            group = explore_source_button,
+            child = new Gtk.Image.from_icon_name ("map-tile-transit") {
+                pixel_size = 48
+            }
+        };
+        transit_source_button.add_css_class ("image-button");
+
+        var transit_source_label = new Gtk.Label (_("Transit")) {
+            mnemonic_widget = transit_source_button
+        };
+        transit_source_label.add_css_class (Granite.STYLE_CLASS_SMALL_LABEL);
+
+        var transit_source_box = new Gtk.Box (VERTICAL, 0);
+        transit_source_box.append (transit_source_button);
+        transit_source_box.append (transit_source_label);
+
+        var source_box = new Gtk.Box (HORIZONTAL, 12) {
+            halign = CENTER,
+            homogeneous = true
+        };
+        source_box.append (explore_source_box);
+        source_box.append (transit_source_box);
+
+        var map_source_item = new MenuItem (null, null);
+        map_source_item.set_attribute_value ("custom", "source");
```

Comment:
Maybe we should leave a comment why we do this?

> For this to work, the menu model of this must have an item with a `custom` attribute that matches id .

https://valadoc.org/gtk4/Gtk.PopoverMenu.add_child.html

Reply:
Ah sorry, I just didn't know this rule out of lack of my knowledge and thus thought leaving comment here

Reply:
So, I'm fine without leaving comment 👍